components/micro-gateway-core/src/main/ballerina/src/gateway/handlers/token_introspection_handler.bal

```diff
@@ -95,14 +96,37 @@ public type KeyValidationHandler object {
                    [authenticationContext, isAllowed] =
                      validateSubscriptionFromDataStores(credential, clientId, apiName, apiVersion,
                      self.validateSubscriptions);
-                    authenticationContext.username = principal?.username ?: USER_NAME_UNKNOWN;
-                    invocationContext.attributes[AUTHENTICATION_CONTEXT] = authenticationContext;
-                    invocationContext.attributes[KEY_TYPE_ATTR] = authenticationContext.keyType;
+                   authenticationContext.username = principal?.username ?: USER_NAME_UNKNOWN;
+                   invocationContext.attributes[AUTHENTICATION_CONTEXT] = authenticationContext;
+                   invocationContext.attributes[KEY_TYPE_ATTR] = authenticationContext.keyType;
+                   if (isAllowed) {
+                       map<string> apiDetails = createAPIDetailsMap(invocationContext);
```

Comment:
Move this method as we dont need this unless the backend jwt generation is enabled.